online-shop/src/app/product-detail/components/edit-product/edit-product.component.ts

```diff
@@ -64,6 +57,13 @@ export class EditProductComponent {
       ],
     });
   }
+  ngOnInit(): void {
+    this.formGroup.get('name')?.setValue(this.product?.name);
```

Comment:
two more ways to do this are:
using form.patchValue
initializing the fields with the right values when you create the form controls. highly likely to have to initialize the form in ngOnInit for this one, instead of the constructor.